samples/csharp_dotnetcore/06.using-cards/Dialogs/MainDialog.cs

```diff
@@ -72,66 +77,79 @@ private async Task<DialogTurnResult> ShowCardStepAsync(WaterfallStepContext step
+                    break;
+                case "Thumbnail Card":
+                    // Display a ThumbnailCard.
+                    reply.Attachments.Add(Cards.GetThumbnailCard().ToAttachment());
+                    break;
+                case "Video Card":
+                    // Display a VideoCard
+                    reply.Attachments.Add(Cards.GetVideoCard().ToAttachment());
+                    break;
+                default:
+                    // Display a carousel of all the rich card types.
+                    reply.AttachmentLayout = AttachmentLayoutTypes.Carousel;
+                    reply.Attachments.Add(Cards.CreateAdaptiveCardAttachment());
+                    reply.Attachments.Add(Cards.GetAnimationCard().ToAttachment());
+                    reply.Attachments.Add(Cards.GetAudioCard().ToAttachment());
+                    reply.Attachments.Add(Cards.GetHeroCard().ToAttachment());
+                    reply.Attachments.Add(Cards.GetReceiptCard().ToAttachment());
+                    reply.Attachments.Add(Cards.GetSigninCard().ToAttachment());
+                    reply.Attachments.Add(Cards.GetThumbnailCard().ToAttachment());
+                    reply.Attachments.Add(Cards.GetVideoCard().ToAttachment());
+                    break;
             }
 
             // Send the card(s) to the user as an attachment to the activity
             await stepContext.Context.SendActivityAsync(reply, cancellationToken);
 
             // Give the user instructions about what to do next
-            await stepContext.Context.SendActivityAsync(MessageFactory.Text("Type anything to see another card."), cancellationToken);
+            await stepContext.Context.SendActivityAsync("Type anything to see another card.", cancellationToken: cancellationToken);
 
             return await stepContext.EndDialogAsync();
         }
+
+        private IList<Choice> GetChoices()
+        {
+            var cardOptions = new List<Choice>()
+            {
+                new Choice() { Value = "Adaptive Card", Synonyms = new List<string>() { "adaptive" } },
+                new Choice() { Value = "Animation Card", Synonyms = new List<string>() { "animation" } },
+                new Choice() { Value = "Audio Card", Synonyms = new List<string>() { "audio" } },
+                new Choice() { Value = "Hero Card", Synonyms = new List<string>() { "hero" } },
+                new Choice() { Value = "Receipt Card", Synonyms = new List<string>() { "receipt" } },
+                new Choice() { Value = "Signin Card", Synonyms = new List<string>() { "signin" } },
+                new Choice() { Value = "Thumbnail Card", Synonyms = new List<string>() { "thumbnail", "thumb" } },
+                new Choice() { Value = "Video Card", Synonyms = new List<string>() { "video" } },
+                new Choice() { Value = "All cards", Synonyms = new List<string>() { "all" } },
+            };
+
+            return cardOptions;
+        }
     }
 }
```

Comment:
really i don't understand why we are touching the C#?
it was fine - the goal here was the exactly port this to JavaScript

Reply:
ok changes make sense - actually a nice improvement - I'd agree it was essentially a bug before